crates/alloy/src/rpc/request.rs

```diff
@@ -52,13 +52,22 @@ pub struct TempoTransactionRequest {
     /// Required when key_type is WebAuthn to calculate calldata gas costs.
     pub key_data: Option<Bytes>,
 
+    /// Whether the signature is a Keychain wrapper (adds 3,000 gas overhead for key validation).
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub is_keychain: bool,
```

Comment:
Is there any way to infer this rather than having the user set this directly? It seems like it could be better UX if the protcol could infer.